src/missing_daemon.c

```diff
@@ -0,0 +1,56 @@
+#include <stdlib.h>
+#include <unistd.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <signal.h>
+
+int missing_daemon(int nochdir, int noclose)
```

Comment:
A few comments about this function in general.

1. the usual order is `chdir()`, then `open()/dup2()`, then `fork()`, `setsid()`, and `fork()` again
2. changing the `umask` is non-standard behavior
3. the `signal()` is also non-standard behavior
4. I would open `/dev/null` as `O_RDWR` and then `dup2()` it onto `0`, `1`, and `2`, before closing the original `fd`. Note the `dup2()` saves you from having to close the fd's 0-2.

Not sure I see the point of using an alternate name if the original name is absent, since there's no chance of a clash by definition.

I would, however, protect the entire function with `#ifndef HAVE_DAEMON` so it compiles down to zero bytes and doesn't pollute the namespace if `HAVE_DAEMON` is defined.

Reply:
Thanks for the review comments!  For the record I cobbled this PR together out of the patch that was submitted in issue #369.

After having looked more carefully at the patch plus your review, I wonder if it'd be better to:

1.  Fix the actual `missing_daemon()` implementation as per your comments.  I had assumed that there was a single canonical order of operations, and that was it.

2.  Rename to the original function name.

3.  Move the function definition to `iperf_util.c` (doesn't need to be in a seperate file) and wrap it with `#ifndef HAVE_DAEMON`.

4.  Maybe add a function declaration to `iperf_util.h`, wrapped by `#ifndef HAVE_DAEMON`.  This might require that `iperf_util.h` include `iperf_config.h`, which it doesn't currently, and I think that might be an issue for the `getline()` definition recently added to the file.

Reply:
Yes to all.